File: An II/Sem I/MAP/Seminar/Sem1/src/Model/SortingTask.java
package Model;

import java.util.Arrays;
import Enums.SortStrategy;

public class SortingTask extends Task{
    private AbstractSorter abstractSorter;
    private final int[] numbers;
    private final SortStrategy sortStrategy;

    public SortingTask(String taskID, String descriere, int[] numbers, SortStrategy sortStrategy) {
        super(taskID, descriere);
        this.numbers = numbers;
        this.sortStrategy = sortStrategy;
        switch (this.sortStrategy) {
            case SortStrategy.BUBBLESORT -> {
                abstractSorter = new BubbleSort();
            }
            case SortStrategy.QUICKSORT -> {
                abstractSorter = new QuickSort();
            }
        }
    }

    @Override
    public void execute() {
        abstractSorter.sort(numbers);
        System.out.println(Arrays.toString(numbers) + '\n');
    }
}

File: An II/Sem I/MAP/Lab/Lab11/Lab11/Model/SortingTask.cs
using Lab11.Enums;

namespace Lab11;

public class SortingTask : Task
{
    private AbstractSorter AbstractSorter { get; set; }
    private SortStrategy sortStrategy { get; set; }
    private int[] elements { get; set; }
    
    public SortingTask(String taskID, String descriere, SortStrategy sortStrategy, int[] elements) : base(taskID, descriere)
    {
        this.sortStrategy = sortStrategy;
        this.elements = elements;
        switch (sortStrategy)
        {
            case SortStrategy.BUBBLESORT:
                this.AbstractSorter = new BubbleSort();
                break;
            case SortStrategy.QUICKSORT:
                this.AbstractSorter = new QuickSort();
                break;
        }
    }
    
    
    public override void execute()
    {
        AbstractSorter.sort(elements);
        Console.WriteLine("Sorted elements: " + string.Join(", ", elements));
    }
}